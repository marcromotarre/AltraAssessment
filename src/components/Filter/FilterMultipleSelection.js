import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const FilterMultipleSelectionContainer = styled.section`
  width: 100%;
  display: flex;
`;

const Title = styled.label`
  grid-area: m-title;
  font-family: 'HelveticaNeue-Thin';
  color: #000;
  font-size: 14px;
  white-space: nowrap;
  padding-right: 10px;
`;

const Elements = styled.section`
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;
const FilterMultipleSelection = ({ title, children, clickOnElement }) => {
  const handleClickOnChild = (event) => {
    clickOnElement(event.currentTarget.id);
  };

  return (
    <FilterMultipleSelectionContainer>
      <Title>{title}:</Title>
      <Elements>
        {children.map((child, index) => (
          <button
            data-testid={`${title}-button-${index}`}
            id={index}
            key={index}
            onClick={handleClickOnChild}
          >
            {child}
          </button>
        ))}
      </Elements>
    </FilterMultipleSelectionContainer>
  );
};

FilterMultipleSelection.propTypes = {
  title: PropTypes.string.isRequired,
  clickOnElement: PropTypes.func,
};

FilterMultipleSelection.defaultProps = {
  clickOnElement: () => {},
};

export default FilterMultipleSelection;
