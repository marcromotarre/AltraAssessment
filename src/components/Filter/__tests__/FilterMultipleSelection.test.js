import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import FilterMultipleSelection from '../FilterMultipleSelection';

/** this component should:
 *  - render FilterMultipleSelection with:
 *      - set a button to each child
 *              - call prop clickOnElement when the button is clicked function receive element index as paramater
 */

const props = {
  title: 'title',
  clickOnElement: jest.fn(),
};

describe('FilterMultipleSelection Component tests', () => {
  it('FilterMultipleSelection Snapshot', () => {
    const tree = renderer
      .create(
        <FilterMultipleSelection {...props}>
          {['div', 'div', 'div'].map((element, index) => (
            <div key={index}></div>
          ))}
        </FilterMultipleSelection>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('FilterMultipleSelection when click on a button call prop clickOnElement with index value', () => {
    const { getByTestId } = render(
      <FilterMultipleSelection {...props}>
        {['div', 'div', 'div'].map((element, index) => (
          <div key={index}></div>
        ))}
      </FilterMultipleSelection>
    );
    expect(props.clickOnElement).toHaveBeenCalledTimes(0);
    const button0 = getByTestId('title-button-0');
    fireEvent.click(button0);
    expect(props.clickOnElement).toHaveBeenCalledTimes(1);
    expect(props.clickOnElement).toHaveBeenLastCalledWith('0');
    const button1 = getByTestId('title-button-1');
    fireEvent.click(button1);
    expect(props.clickOnElement).toHaveBeenCalledTimes(2);
    expect(props.clickOnElement).toHaveBeenLastCalledWith('1');
    const button2 = getByTestId('title-button-2');
    fireEvent.click(button2);
    expect(props.clickOnElement).toHaveBeenCalledTimes(3);
    expect(props.clickOnElement).toHaveBeenLastCalledWith('2');
  });
});
