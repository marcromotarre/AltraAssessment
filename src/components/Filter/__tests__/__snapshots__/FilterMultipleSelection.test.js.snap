// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterMultipleSelection Component tests FilterMultipleSelection Snapshot 1`] = `
<section
  className="sc-AxjAm gTqIvE"
>
  <label
    className="sc-AxirZ emtTfs"
  >
    title
    :
  </label>
  <section
    className="sc-AxiKw cmRZTH"
  >
    <button
      data-testid="title-button-0"
      id={0}
      onClick={[Function]}
    >
      <div />
    </button>
    <button
      data-testid="title-button-1"
      id={1}
      onClick={[Function]}
    >
      <div />
    </button>
    <button
      data-testid="title-button-2"
      id={2}
      onClick={[Function]}
    >
      <div />
    </button>
  </section>
</section>
`;
